app/services/hosts_sync.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Synchronizes the hosts in our database with the hosts in the Inventory API
+class HostsSync
+  class << self
+    def mark_last_seen
+      Account.all.each do |account|
+        logger.info "Mark job: #{account.account_number}"
+        account.hosts.where(
+          'last_seen_in_inventory <= ?', 1.day.ago
+        ).find_in_batches(batch_size: 100).map do |hosts|
+          check_hosts_in_inventory(hosts, account)
+        end
+      end
+    end
+
+    def remove_disabled_hosts
+      disabled_hosts = Host.where(disabled: true,
+                                  last_seen_in_inventory: 1.day.ago)
+      logger.info 'Sweep job: hosts about to be removed - '\
+        "#{disabled_hosts.pluck(:id)}"
+      RuleResult.where(host_id: disabled_hosts.pluck(:id)).destroy_all
```

Comment:
Hmm... I don't think this is necessary since we have the `dependent: :delete_all` on the host. Right?

Reply:
@akofink It doesn't seem to work, it still `destroys all` - making a ton of queries